src/App.jsx

```diff
@@ -14,33 +15,93 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+export const App = () => {
+  const [goods, setGoods] = useState(goodsFromServer);
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortType, setSortType] = useState(null);
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  const handleSort = type => {
+    let sortedGoods = [...goodsFromServer];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+    if (type === 'alphabet') {
+      sortedGoods = sortedGoods.sort();
+    } else if (type === 'length') {
+      sortedGoods = sortedGoods.sort((a, b) => a.length - b.length);
+    }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+    if (isReversed) {
+      sortedGoods.reverse();
+    }
+
+    setGoods(sortedGoods);
+    setSortType(type);
+  };
+
+  const handleReverse = () => {
+    const reversedGoods = [...goods].reverse();
+
+    setGoods(reversedGoods);
+
+    if (JSON.stringify(reversedGoods) === JSON.stringify(goodsFromServer)) {
+      setSortType(null);
+    }
+
+    setIsReversed(!isReversed);
```

Comment:
Use functional `setState` when you need to change state depending on the previous one